internal/filter/filterconfig/config_test.go

```diff
@@ -2,3 +2,209 @@
 // SPDX-License-Identifier: Apache-2.0
 
 package filterconfig
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterset/regexp"
+)
+
+func Test_ValidateWithSpans(t *testing.T) {
+	tests := []struct {
+		name    string
+		config  *MatchProperties
+		wantErr bool
+	}{
+		{
+			name: "valid",
+			config: &MatchProperties{
+				SpanKinds: []string{"kind"},
+				Config: filterset.Config{
+					MatchType: filterset.Regexp,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "invalid",
+			config: &MatchProperties{
+				SpanKinds: []string{"kind"},
+				Config: filterset.Config{
+					MatchType: filterset.Regexp,
+				},
+				MetricNames: []string{"name"},
+			},
+			wantErr: true,
+		},
+		{
+			name:    "invalid empty",
+			config:  &MatchProperties{},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.wantErr {
+				require.Error(t, tt.config.ValidateForSpans())
+			} else {
+				require.NoError(t, tt.config.ValidateForSpans())
+			}
+		})
+	}
+}
+
+func Test_ValidateWithMetrics(t *testing.T) {
+	tests := []struct {
+		name    string
+		config  *MatchProperties
+		wantErr bool
+	}{
+		{
+			name: "valid",
+			config: &MatchProperties{
+				MetricNames: []string{"kind"},
+				Config: filterset.Config{
+					MatchType: filterset.Regexp,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "invalid",
```

Comment:
Ideally I think we should have an invalid test for each property we don't allow.